backend/LeetWars.Builder/Services/SolutionRunnerService.cs

```diff
@@ -1,29 +1,129 @@
 ﻿using Docker.DotNet;
 using Docker.DotNet.Models;
+using LeetWars.Builder.Helpers.BuildResultReader;
+using LeetWars.Builder.Helpers.DirectoryConfigurations;
+using LeetWars.Builder.Helpers.DockerConfigurations;
 using LeetWars.Builder.Interfaces;
+using LeetWars.Builder.Models;
+using Newtonsoft.Json;
 
 namespace LeetWars.Builder.Services
 {
     public class SolutionRunnerService : ISolutionRunnerService, IDisposable
     {
         private readonly DockerClient _client = new DockerClientConfiguration().CreateClient();
 
-        public async Task RunCodeInContainerAsync(string imageName, string containerName)
+        public async Task<CodeRunResults> Run(CodeRunRequest request)
         {
-            var containerParams = new CreateContainerParameters
+            var reader = new BuildResultReader();
+            var dirBuilder = new DirectoryConfigurations();
+            var dockerConfig = new DockerConfigurations();
+
+            var dir = Directory.GetCurrentDirectory();
+            var solutionFileName = "";
+
+            var config = new Config();
+
+            if(request.Language == "C#")
+            {
+                var projectFileName = await dirBuilder.CreateDirectoryCSharp(dir, request);
+
+                var imagePullParams = new ImagesCreateParameters
+                {
+                    FromImage = "mcr.microsoft.com/dotnet/sdk:6.0"
+                };
+
+                var progress = new Progress<JSONMessage>();
+                await _client.Images.CreateImageAsync(imagePullParams, null, progress);
+
+                solutionFileName = "Solution.cs";
+
+                config = dockerConfig.GetCSharpConfig(projectFileName);
+            }
+
+            if(request.Language == "Javascript")
+            {
+                dirBuilder.CreateDirectoryJS(dir, request);
+
+                var imagePullParams = new ImagesCreateParameters
+                {
+                    FromImage = "node:18"
+                };
+
+                var progress = new Progress<JSONMessage>();
+                await _client.Images.CreateImageAsync(imagePullParams, null, progress);
+
+                solutionFileName = "Solution.js";
+
+                config = dockerConfig.GetJSConfig(solutionFileName);
+            }
```

Comment:
make it more readable, try moving out similiar logic

Reply:
maximum what I can move here is image creation, the other just have different implementation